src/data/real/ennreal.lean

```diff
@@ -906,9 +905,10 @@ lemma one_half_lt_one : (2⁻¹:ennreal) < 1 := inv_lt_one.2 $ one_lt_two
 lemma half_lt_self {a : ennreal} (hz : a ≠ 0) (ht : a ≠ ⊤) : a / 2 < a :=
 begin
   lift a to nnreal using ht,
-  norm_cast at *,
-  rw [← coe_div _root_.two_ne_zero'], -- `norm_cast` fails to apply `coe_div`
-  norm_cast,
+  have h : (2 : ennreal) = ((2 : nnreal) : ennreal), from rfl,
+  have h' : (2 : nnreal) ≠ 0, from _root_.two_ne_zero',
+  rw [h, ← coe_div h', coe_lt_coe], -- `norm_cast` fails to apply `coe_div`
+  norm_cast at hz,
```

Comment:
How worried should I be able this apparent regression?

Reply:
Oh, I see, there's a later TODO and you're still working on it.

Reply:
I've fixed the easy todos.  I'd rather merge this PR in an imperfect state and improve it later on.

The core issue is the following: the `coe_div : ((x / y : ℝ≥0) : ennreal) = x / y` lemma requires the assumption `y ≠ 0`.  (This is necessary, because it is wrong otherwise.)  So when `norm_cast` permutes `↑` and `/` it needs to prove that `2 ≠ 0`.

Previously it used `simpa` to prove this, now it uses `assumption`.  I'm not sure what the rationale is here, but this change was mentioned as a big feature in #1103.